### DISCUSSION TITLE
subpartitioning and part_config_sub

### DISCUSSION THREAD

**Discussion by beckom4** [pgpartman/pg_partman#725] [General]
Hi, 
According to the documentation( and to our experience), the run_maintenance() runs on tables that appear in the part_config table only.
That means that when we use pg_partman to manage subpartitions, we have to configure part_config as well as part_config_sub. Is it possible that we missed something? If not, we were wondering what the role of part_config_sub is if run_maintneance takes its data from the part_config table anyways. 
Thanks!

**Comment by keithf4** [comment]
When sub-partitioning, the child tables which are themselves parent tables of another partition set, may have a different partition configuration than the upper level parent. Like at the top level it's partitioned by month, but the child tables are themselves partitioned by 100,000 integer intervals or daily or some other interval. `part_config_sub` stores that information so that when a new child table is created in a partition set, pg_partman knows that it itself is also a parent table of another partition set and creates it using the information from `part_config_sub`.
